viz/viz_factory.py

```diff
@@ -14,25 +14,53 @@ class VizFactory:
         vc.STACKED_BAR_CHART: vb.StackedBarChart
     }
 
-    def render_viz(self, chart_type, headers, n_sizes, iteration_data, reverse):
-
-        # withdraw first iteration to form backbone for processed_lines
-        # then add each pcnt to the corresponding backbone entry for each extra iteration
-        baseline_data = iteration_data[0]
-        percentages, fn_names, fn_descriptors = self.extract_base_functions(baseline_data)
-        gradient = self._generate_gradient(percentages)
-        processed_lines = [([pcnt], fn, grad) for pcnt, fn, grad in
-                           zip(percentages, fn_names, gradient)]
-
-        for idx in range(1, len(iteration_data)):
-            line_percentages = self.extract_iteration_percentages(iteration_data[idx], headers, fn_descriptors)
-            for i, pcnt in enumerate(line_percentages):
-                processed_lines[i][0].append(pcnt)
-
-        if reverse:
-            self._reverse_entries(n_sizes, processed_lines)
-
-        builder = self.FACTORY[chart_type](processed_lines, n_sizes)
+    def render_viz(self, chart_type, headers, n_sizes, iteration_data, render_options):
+        print(render_options)
+        reverse = render_options["reverse"]
+        track = render_options["track"]
+
+        if not track:
+            chart_data = []
+            all_fns = []
```

Comment:
this should be a set if you aren't going to use it as an array
```suggestion
            all_fns = set()
```